src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceConnection.cs

```diff
@@ -18,7 +18,8 @@ namespace Microsoft.Azure.SignalR.AspNet
     internal partial class ServiceConnection : ServiceConnectionBase
     {
         private const string ReconnectMessage = "asrs:reconnect";
-        private readonly ConcurrentDictionary<string, AzureTransport> _clientConnections = new ConcurrentDictionary<string, AzureTransport>(StringComparer.Ordinal);
+        private readonly ConcurrentDictionary<string, IServiceTransport> _clientConnections = new ConcurrentDictionary<string, IServiceTransport>(StringComparer.Ordinal);
+        private readonly ConcurrentDictionary<string, Channel<ServiceMessage>> _clientChannels = new ConcurrentDictionary<string, Channel<ServiceMessage>>();
```

Comment:
Why do we have 2 dictionaries instead of storing the channel as state on the transport?

Reply:
Because we put create connection message in channel, too. And transport is not created at that time.